stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -234,7 +229,9 @@ struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType, M,
     save_varis(dest + x.size(), args...);
   }
 
-  template <typename R, typename... Pargs>
+  template <typename R,
+	    require_arithmetic_t<scalar_type_t<R>>...,
```

Comment:
C++ question, but this didn't work when I had just:

```
require_arithmetic_t<scalar_type_t<R>>,
```

Out of desparation I added the ...s and it went to working. Why? R isn't a parameter pack?

Reply:
Yeah sorry the `requires` uses a weird thing where `require_t<>...` is actually constructing a parameter pack and not expanding one. If `require_arithmetic_t<scalar_type_t<R>>` is valid this makes an empty parameter pack that gets ignored but otherwise fails SFINAE